src/main/java/com/mraof/minestuck/block/MSBlocks.java

```diff
@@ -290,7 +291,7 @@ private static Block coalOre(BlockBehaviour.Properties properties)
 	public static final RegistryObject<Block> SHADEWOOD_LEAVES = REGISTER.register("shadewood_leaves", () -> new FlammableLeavesBlock(Block.Properties.of(Material.LEAVES).strength(0.2F).randomTicks().sound(SoundType.GRASS).noOcclusion().isValidSpawn(MSBlocks::leafSpawns).isSuffocating(MSBlocks::never).isViewBlocking(MSBlocks::never)));
 	public static final RegistryObject<Block> SHROOMY_SHADEWOOD_LEAVES = REGISTER.register("shroomy_shadewood_leaves", () -> new FlammableLeavesBlock(Block.Properties.of(Material.LEAVES).strength(0.2F).randomTicks().lightLevel(state -> 11).sound(SoundType.GRASS).noOcclusion().isValidSpawn(MSBlocks::leafSpawns).isSuffocating(MSBlocks::never).isViewBlocking(MSBlocks::never)));
 	
-	public static final RegistryObject<BushBlock> RAINBOW_SAPLING = REGISTER.register("rainbow_sapling", () -> new RainbowSaplingBlock(Block.Properties.of(Material.PLANT).noCollission().randomTicks().strength(0).sound(SoundType.GRASS)));
+	public static final RegistryObject<BushBlock> RAINBOW_SAPLING = REGISTER.register("rainbow_sapling", () -> new SaplingBlock(new RainbowTree(), Block.Properties.of(Material.PLANT).noCollission().randomTicks().strength(0).sound(SoundType.GRASS)));
```

Comment:
The only issue I have with this is that the SaplingBlock cannot survive on wool or terracotta. The Feature that places the rainbow trees checks to see whether a rainbow sapling would be capable of being placed on the ground and since theres never a dirt material there for it to pick from, there ends up being no trees in the land. You may need to reimplement RainbowSaplingBlock and make it match closer to AspectSaplingBlock or EndSaplingBlock

Reply:
Ah right, I forgot about that when I originally suggested replacing the block class with `SaplingBlock`. But yeah, the solution is to reintroduce `RainbowSaplingBlock`, but extending `SaplingBlock` and only overriding whatever it needs to change the supported ground block.